src/main/java/nextstep/session/service/SessionService.java

```diff
@@ -0,0 +1,86 @@
+package nextstep.session.service;
+
+import nextstep.session.domain.Session;
+import nextstep.session.domain.SessionCoverImage;
+import nextstep.session.domain.SessionRepository;
+import nextstep.session.domain.student.SessionStudent;
+import nextstep.session.domain.student.SessionStudentRepository;
+import nextstep.session.domain.student.SessionStudents;
+import nextstep.users.domain.NsUser;
+import nextstep.users.domain.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class SessionService {
+
+    private final SessionRepository sessionRepository;
+    private final SessionStudentService studentService;
+    //private final SessionTeacherService sessionTeacherService;
+
+    public SessionService(
+            SessionRepository jdbcSessionRepository, SessionStudentService sessionStudentService
+    ) {
+        this.sessionRepository = jdbcSessionRepository;
+        this.studentService = sessionStudentService;
+    }
+    @Transactional
+    public Long enrollStudent(NsUser studentUser, Long sessionId) {
+        Session session = sessionRepository.findById(sessionId).orElseThrow(() -> new IllegalArgumentException("존재하지 않는 강의입니다."));
+        SessionStudents studentsOfSession = studentService.getStudentsOfSession(session);
+        Session session1 = new Session(session, studentsOfSession);
+        return studentService.enrollStudent(session1, studentUser);
```

Comment:
```suggestion
        SessionStudent student = session.enroll(studentsOfSession, studentUser);
        return studentService.enroll(student);
```
위와 같은 구조로 구현하는 것은 어떨까?
Session을 두 번 생성하다보니 변수명을 어쩔 수 없이 session1으로 구현하는 것도 좋은 방법은 아니라는 생각이 든다.